src/test/java/games/strategy/engine/data/UnitCollectionTest.java

```diff
@@ -139,52 +125,44 @@ public void unitCollection() {
-    unitCollection.addUnit(unitDefaultPlayer);
+    unitCollection.add(unitDefaultPlayer);
 
     assertThat(unitCollection.getUnitCount(), is(equalTo(1)));
     assertThat(unitCollection.getUnits().iterator().next(), is(equalTo(unitDefaultPlayer)));
-    final int expNotifyChangedCountAfter = notifyChangedCountBefore + 1;
-    assertThat(getDefaultPlayerNotifyChangedCounter(), is(equalTo(expNotifyChangedCountAfter)));
+    verify(defaultPlayerId).notifyChanged();
   }
 
   @Test
-  public void addAllUnitsFromCollection() {
-    final int notifyChangedCountBefore = getDefaultPlayerNotifyChangedCounter();
+  public void addAllFromCollection() {
     final Collection<Unit> unitsOfOtherPlayerOfUnitTypeOne = getOtherPlayerUnitsOfUnitTypeOne();
-    unitCollection.addAllUnits(unitsOfOtherPlayerOfUnitTypeOne);
+    unitCollection.addAll(unitsOfOtherPlayerOfUnitTypeOne);
 
     assertThat(unitCollection.getUnitCount(), is(equalTo(unitsOfOtherPlayerOfUnitTypeOne.size())));
-    final int expNotifyChangedCountAfter = notifyChangedCountBefore + 1;
-    assertThat(getDefaultPlayerNotifyChangedCounter(), is(equalTo(expNotifyChangedCountAfter)));
+    verify(defaultPlayerId).notifyChanged();
   }
 
   @Test
-  public void addAllUnitsFromUnitCollection() {
-    final int notifyChangedCountBefore = getDefaultPlayerNotifyChangedCounter();
+  public void addAllFromUnitCollection() {
     final Collection<Unit> unitsOfOtherPlayerOfUnitTypeOne = getOtherPlayerUnitsOfUnitTypeOne();
     final UnitCollection unitCollectionTwo = new UnitCollection(otherPlayerId, mockGameData);
-    unitCollectionTwo.addAllUnits(unitsOfOtherPlayerOfUnitTypeOne);
-    unitCollection.addAllUnits(unitCollectionTwo);
+    unitCollectionTwo.addAll(unitsOfOtherPlayerOfUnitTypeOne);
+    unitCollection.addAll(unitCollectionTwo);
 
     assertThat(unitCollection.getUnitCount(), is(equalTo(unitsOfOtherPlayerOfUnitTypeOne.size())));
-    final int expNotifyChangedCountAfter = notifyChangedCountBefore + 1;
-    assertThat(getDefaultPlayerNotifyChangedCounter(), is(equalTo(expNotifyChangedCountAfter)));
+    verify(defaultPlayerId).notifyChanged();
   }
 
   @Test
-  public void removeAllUnits() {
+  public void removeAll() {
     final Collection<Unit> unitsOfOtherPlayerOfUnitTypeOne = getOtherPlayerUnitsOfUnitTypeOne();
-    unitCollection.addAllUnits(unitsOfOtherPlayerOfUnitTypeOne);
+    unitCollection.addAll(unitsOfOtherPlayerOfUnitTypeOne);
 
-    final int notifyChangedCountBefore = getDefaultPlayerNotifyChangedCounter();
-    unitCollection.removeAllUnits(unitsOfOtherPlayerOfUnitTypeOne);
+    unitCollection.removeAll(unitsOfOtherPlayerOfUnitTypeOne);
 
     assertThat(unitCollection.getUnitCount(), is(equalTo(0)));
-    final int expNotifyChangedCountAfter = notifyChangedCountBefore + 1;
-    assertThat(getDefaultPlayerNotifyChangedCounter(), is(equalTo(expNotifyChangedCountAfter)));
+    verify(defaultPlayerId, times(2)).notifyChanged();
```

Comment:
In this case, the first invocation is a side effect of the "arrange/given" part of this test.  I don't like using `Mockito#reset()` in general, but this seems like it might be more appropriate than possibly misleading a future test reader that two calls to the handler are expected.  For example:

```java
@Test
public void removeAll() {
  final Collection<Unit> unitsOfOtherPlayerOfUnitTypeOne = getOtherPlayerUnitsOfUnitTypeOne();
  unitCollection.addAll(unitsOfOtherPlayerOfUnitTypeOne);
  reset(defaultPlayerId); // <-- ignore handler call from addAll()

  unitCollection.removeAll(unitsOfOtherPlayerOfUnitTypeOne);

  assertThat(unitCollection.getUnitCount(), is(equalTo(0)));
  verify(defaultPlayerId).notifyChanged(); // <-- verify only one call to handler from removeAll()
}
```

This [SE answer](https://softwareengineering.stackexchange.com/a/188604/189282) suggests that this type of `reset()` usage is ok, but that it would be preferable to have its use hidden in a separate method responsible for setting up the test, e.g. `givenUnitCollectionWithUnitsOfOtherPlayerOfUnitTypeOne()`.  That may not be necessary in this one case given that the arrange/act/assert parts of the test are already delineated by whitespace.
```